lib/Driver/Driver.cpp

```diff
@@ -2181,6 +2181,12 @@ static void addAuxiliaryOutput(Compilation &C, CommandOutput &output,
     // multiple primary inputs; really it should be associated with the build as
     // a whole -- derived OFM input "" -- but that's a more general thing to
     // fix.
+    if (requireArg.isValid() && !C.getArgs().getLastArg(requireArg)) {
```

Comment:
This code looks out of place to me. How about folding this into the previous else on line 2173. In other words 2173-2188 would be:
````
else if (requreArg.isValid() ...) {
  ...
  return
} else {
  // Put the auxilliary file next to ....
````